### PR TITLE
make dns server configurable for user containers

### PR DESCRIPTION
**Pull request by jeremiaswerner**

### CODE REVIEW

**Comment by jeremiaswerner** [comment]
travis with default, i.e. no DNS server
https://travis-ci.org/openwhisk/openwhisk/builds/231312555?utm_source=github_status&utm_medium=notification

File: core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala [hunk 1]
```diff
@@ -73,6 +75,8 @@ object DockerContainer {
             case (key, value) => Seq("-e", s"$key=$value")
         }.flatten
 
+        val dnsArgs = dnsServer.foldLeft(Array.empty[String]) { (a,d) => a ++ Array("--dns", d) }
```
Comment (markusthoemmes) [inline core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala]: Should be equivalent to:

```scala
dnsServers.map(Seq("--dns", _)).flatten
```

(I'd always prefer the more abstract `map` over a relatively lowlevel `foldLeft`).

Also note that `Array` is a mutable/indexed datastructure, thus it's usage should be chosen carefully.

File: core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala [hunk 2]
```diff
@@ -65,6 +66,7 @@ object DockerContainer {
                cpuShares: Int = 0,
                environment: Map[String, String] = Map(),
                network: String = "bridge",
+               dnsServer: Seq[String] = Seq(),
```
Comment (markusthoemmes) [inline core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala]: Should we make this plural, as its a `Seq`?

**Comment by markusthoemmes** [comment]
Thanks for also updating the new containerpool bits :+1:

**Comment by jeremiaswerner** [comment]
PG1-1537 🔵

**Pull request merged**